tools/ci/appveyor/numpy.patch

```diff
@@ -0,0 +1,825 @@
++                'extra_f77_compile_args') or []
++            fcompiler.extra_f90_compile_args = build_info.get(
++                'extra_f90_compile_args') or []
+ 
+         macros = build_info.get('macros')
+         include_dirs = build_info.get('include_dirs')
+@@ -203,9 +261,10 @@ class build_clib(old_build_clib):
+         # where compiled F90 module files are:
+         module_dirs = build_info.get('module_dirs') or []
+         module_build_dir = os.path.dirname(lib_file)
+-        if requiref90: self.mkpath(module_build_dir)
++        if requiref90:
++            self.mkpath(module_build_dir)
+ 
+-        if compiler.compiler_type=='msvc':
++        if compiler.compiler_type == 'msvc':
+             # this hack works around the msvc compiler attributes
+             # problem, msvc uses its own convention :(
+             c_sources += cxx_sources
+@@ -239,7 +298,7 @@ class build_clib(old_build_clib):
+             if requiref90:
+                 if fcompiler.module_dir_switch is None:
+                     existing_modules = glob('*.mod')
+-                extra_postargs += fcompiler.module_options(\
++                extra_postargs += fcompiler.module_options(
+                     module_dirs, module_build_dir)
+ 
+             if fmodule_sources:
+@@ -257,14 +316,14 @@ class build_clib(old_build_clib):
+                     if f in existing_modules:
+                         continue
+                     t = os.path.join(module_build_dir, f)
+-                    if os.path.abspath(f)==os.path.abspath(t):
++                    if os.path.abspath(f) == os.path.abspath(t):
+                         continue
+                     if os.path.isfile(t):
+                         os.remove(t)
+                     try:
+                         self.move_file(f, module_build_dir)
+                     except DistutilsFileError:
+-                        log.warn('failed to move %r to %r' \
++                        log.warn('failed to move %r to %r'
+                                  % (f, module_build_dir))
+ 
+             if f_sources:
+@@ -278,11 +337,49 @@ class build_clib(old_build_clib):
+         else:
+             f_objects = []
+ 
+-        objects.extend(f_objects)
+-
+-        # assume that default linker is suitable for
+-        # linking Fortran object files
+-        compiler.create_static_lib(objects, lib_name,
++        libs = []
++
++        if objects:
++            # Go ahead and compile the c code
++            compiler.create_static_lib(objects, lib_name + '_msvc',
++                                       output_dir=self.build_clib,
++                                       debug=self.debug)
++            libs += [os.path.join(self.build_clib, lib_name + '_msvc.lib')]
++
++        if f_objects:
++            if lib_name == 'odepack':
++                f_objects = [
++                    obj for obj in f_objects if 'xsetf' not in obj and 'xsetun' not in obj]
```

Comment:
Sorry for my ignorance - but does it make sense to delete these objects from `odepack`?  Are they not necessary?  Or are they duplicates?